python/Ganga/GPIDev/Base/VPrinter.py

```diff
@@ -264,36 +274,39 @@ def full_print(obj, out=None, interactive=False):
     if out == None:
         out = sys.stdout
 
-    from Ganga.GPIDev.Lib.GangaList import GangaList
+    from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
 
     _obj = stripProxy(obj)
 
-    if isType(_obj, GangaList.GangaList):
+    if isType(_obj, GangaList):
         obj_len = len(_obj)
         if obj_len == 0:
             print('[]', end=' ', file=out)
         else:
             outString = '['
-            count = 0
-            for x in obj:
-                if isType(x, GangaObject):
+            outStringList = []
+            def print_x(x, outStringList, obj_len):
+                if isinstance(x, (str, int, float, bool, type)):
+                    outStringList.append(str(x).rstrip())
+                elif isType(x, GangaObject):
                     sio = StringIO()
                     stripProxy(x).printTree(sio, interactive)
                     result = sio.getvalue()
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
+                    outStringList.append(result.rstrip())
                 else:
-                    result = str(x)
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
-                count += 1
-                if count != obj_len:
-                    outString += ', '
+                    outStringList.append(str(x).rstrip())
```

Comment:
The `else` is the same as the first `if`. I can't think of a reason to add in the special case at the beginning?